internal/tools/aws/database_multitenant_pgbouncer.go

```diff
@@ -395,36 +398,35 @@ func (d *RDSMultitenantPGBouncerDatabase) provisionPGBouncerDatabase(vpcID strin
 func (d *RDSMultitenantPGBouncerDatabase) ensurePGBouncerAuthUserSecretIsCreated(rdsClusterID, VpcID *string) (*RDSSecret, error) {
 	authUserSecretName := PGBouncerAuthUserSecretName(*VpcID)
 
-	secretValue, err := d.client.Service().secretsManager.GetSecretValue(&secretsmanager.GetSecretValueInput{
-		SecretId: aws.String(authUserSecretName),
-	})
-	if err != nil && !IsErrorCode(err, secretsmanager.ErrCodeResourceNotFoundException) {
+	secret, err := d.client.secretsManagerGetRDSSecret(authUserSecretName, d.client.logger)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get secret value for database")
+	}
+
+	var awsErr *smTypes.ResourceNotFoundException
+	if err != nil && !errors.As(err, &awsErr) {
 		return nil, errors.Wrapf(err, "failed to get pgbouncer auth user secret %s", authUserSecretName)
 	}
 
-	var secret *RDSSecret
-	if secretValue != nil && secretValue.SecretString != nil {
-		secret, err = unmarshalSecretPayload(*secretValue.SecretString)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to unmarshal pgbouncer auth user secret %s", authUserSecretName)
-		}
-	} else {
-		description := RDSMultitenantPGBouncerClusterSecretDescription(*VpcID)
-		tags := []*secretsmanager.Tag{
-			{
-				Key:   aws.String(trimTagPrefix(DefaultRDSMultitenantDatabaseIDTagKey)),
-				Value: rdsClusterID,
-			},
-			{
-				Key:   aws.String(trimTagPrefix(VpcIDTagKey)),
-				Value: VpcID,
-			},
-		}
+	if secret != nil && err != nil {
+		return nil, errors.Wrapf(err, "failed to unmarshal pgbouncer auth user secret %s", authUserSecretName)
+	}
 
-		secret, err = createDatabaseUserSecret(authUserSecretName, DefaultPGBouncerAuthUsername, description, tags, d.client)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to create a multitenant RDS database secret %s", authUserSecretName)
-		}
+	description := RDSMultitenantPGBouncerClusterSecretDescription(*VpcID)
```

Comment:
It will also be optional.

Reply:
What do you mean by optional?

Reply:
If we already have a secret from `secretsManagerGetRDSSecret`, we do not need to create again.